crates/rethnet_evm/benches/state.rs

```diff
@@ -0,0 +1,137 @@
+                c.benchmark_group(format!(
+                    "StateRef::insert_account() with {} accounts with {} layer(s) per account",
+                    *number_of_accounts, layers_per_account,
+                ))
+                .bench_function(label, |b| {
+                    b.iter_batched(
+                        || state_factory(),
+                        |mut state| {
+                            state.insert_account(
+                                Address::from_str(&format!("0x{:0>40x}", number_of_accounts + 1))
+                                    .unwrap(),
+                                AccountInfo::default(),
+                            )
+                        },
+                        BatchSize::LargeInput,
+                    );
+                });
+            }
+        }
+    }
+}
+
+fn bench_checkpoint(c: &mut Criterion) {
+    for number_of_accounts in ADDRESS_SCALES.iter() {
+        for layers_per_account in LAYER_SCALES.iter() {
+            let mut rethnet_states = RethnetStates::default();
+            rethnet_states.fill(*number_of_accounts, *layers_per_account);
+
+            for (label, state_factory) in rethnet_states.make_clone_factories().into_iter() {
+                c.benchmark_group(format!(
+                    "StateDebug::checkpoint() with {} accounts with {} layer(s) per account",
+                    *number_of_accounts, layers_per_account,
+                ))
+                .bench_function(label, |b| {
+                    b.iter_batched(
+                        || state_factory(),
+                        |mut state| state.checkpoint(),
+                        BatchSize::LargeInput,
+                    );
+                });
+            }
+        }
+    }
+}
+
+fn bench_basic(c: &mut Criterion) {
+    for number_of_accounts in ADDRESS_SCALES.iter() {
+        let layers_per_account = 1;
+
+        let mut rethnet_states = RethnetStates::default();
+        rethnet_states.fill(*number_of_accounts, layers_per_account);
+
+        for (label, state_factory) in rethnet_states.make_clone_factories().into_iter() {
+            c.benchmark_group(format!(
+                "StateRef::basic() with {} accounts with {} layer(s) per account",
+                *number_of_accounts, layers_per_account,
+            ))
+            .bench_function(label, |b| {
+                b.iter_batched(
+                    || state_factory(),
+                    |state| {
+                        for i in *number_of_accounts..=1 {
+                            state
+                                .basic(Address::from_str(&format!("0x{:0>40x}", i)).unwrap())
+                                .unwrap();
+                        }
+                    },
+                    BatchSize::LargeInput,
+                )
+            });
+        }
+    }
+}
+
+criterion_group! {
+    name = benches;
+    config = Criterion::default().significance_level(0.1).sample_size(10);
```

Comment:
See if we really need this sample size and significance level modification. We probably only need to do it for particular benchmarks (eg `checkpoint`) rather than ALL of them like it's coded for here.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3868/commits/1c0c5c3b61ab5ac1246e83eb0a702846e27c7f34